lasy/laser.py

```diff
@@ -160,6 +160,46 @@ def normalize(self, value, kind="energy"):
         else:
             raise ValueError(f'kind "{kind}" not recognized')
 
+    def apply_optics(self, optical_element):
+        """
+        Propagate the laser pulse through a thin optical element.
+
+        Parameter
+        ---------
+        optical_element: an :class:`.OpticalElement` object (optional)
+            Represents a thin optical element, through which the laser
+            propagates.
+        """
+        # Transform the field from temporal to frequency domain
+        time_axis_indx = -1
+        field_fft = np.fft.ifft(self.grid.field, axis=time_axis_indx, norm="backward")
+
+        # Create the frequency axis
+        dt = self.grid.dx[time_axis_indx]
+        omega0 = self.profile.omega0
+        Nt = self.grid.field.shape[time_axis_indx]
+        omega = 2 * np.pi * np.fft.fftfreq(Nt, dt) + omega0
+
+        # Apply optical element
+        if self.dim == "rt":
+            r, w = np.meshgrid(self.grid.axes[0], omega, indexing="ij")
```

Comment:
(same as comment in `optical_element.py`: should the `meshgrid` be done in the optical element? Could just be an option)